AssetRipper.GUI/ConsoleApp.cs

```diff
@@ -1,126 +1,179 @@
+			bool verbose = ConsoleOptions.GetOptionOrFallback(ConsoleOptions.verboseOption, false);
+
+			// Get a list of options supported by LibraryConfiguration
+			// MUST be run after defaults have been filled, as it only lists
+			// options with values.
+			foreach (Type type in ripper.Settings.settings.Keys.ToList())
 			{
-				if (MultiFileStream.Exists(arg))
-				{
+				if (type == null)
 					continue;
-				}
 
-				if (Directory.Exists(arg))
-				{
+				string optionName = ConsoleOptions.OptionNameFromType(type);
+				options.TryGetValue(optionName, out Option? option);
+
+				if (option == null)
 					continue;
-				}
 
-				System.Console.WriteLine(MultiFileStream.IsMultiFile(arg)
-					? $"File '{arg}' doesn't have all parts for combining"
-					: $"Neither file nor directory with path '{arg}' exists");
+				object? settingValue = context.ParseResult.GetValueForOption(option);
 
-				return false;
-			}
+				if (settingValue == null)
+					continue;
 
-			try
-			{
-				options.LogFile ??= new FileInfo(ExecutingDirectory.Combine(DefaultLogFileName));
-				options.OutputDirectory ??= new DirectoryInfo(ExecutingDirectory.Combine("Ripped"));
-			}
-			catch (Exception ex)
-			{
-				System.Console.WriteLine($"Failed to initialize the output and log paths.");
-				System.Console.WriteLine(ex.ToString());
-				return false;
+				string? enumKey = settingValue.ToString();
+
+				if (string.IsNullOrEmpty(enumKey))
+					continue;
+
+				ripper.Settings.SetSetting(type, Enum.Parse(type, enumKey));
 			}
 
-			return true;
+			DoRipping(ripper, inputs, output.FullName, verbose, logFile.FullName);
 		}
 
-		private static void Run(Options options)
+		private static void HandleException(Exception ex, InvocationContext context)
 		{
-			Logger.AllowVerbose = options.Verbose;
-			Logger.Add(new ConsoleLogger(false));
-			Logger.Add(new FileLogger(options.LogFile.FullName));
-			Logger.LogSystemInformation("AssetRipper Console Version");
-#if !DEBUG
-			try
-#endif
-			{
-				Ripper ripper = new();
-				ripper.Settings.LogConfigurationValues();
-				ripper.Load(options.FilesToExport);
-				PrepareExportDirectory(options.OutputDirectory.FullName);
-				ripper.ExportProject(options.OutputDirectory.FullName);
-			}
-#if !DEBUG
-			catch (Exception ex)
-			{
-				Logger.Log(LogType.Error, LogCategory.General, ex.ToString());
-			}
+			HandleException(ex);
+		}
+
+		private static void HandleException(Exception ex)
+		{
+			Environment.ExitCode = (int)ExitCode.ExtractingError;
+
+#if DEBUG
+			Console.WriteLine("===================================================");
+			Console.WriteLine("      Error during ripping, extraction aborted     ");
+			Console.WriteLine("===================================================");
+			Console.WriteLine(ex.ToString());
+#else
+			Console.WriteLine("Error during ripping, extraction aborted");
+			Console.WriteLine("See this message below for details:");
+			Console.WriteLine(ex.Message);
```

Comment:
Errors like this must always be saved to the log.

Reply:
@DecentM todo